src/wasm/wasm-validator.cpp

```diff
@@ -237,14 +237,17 @@ void WasmValidator::visitAtomicRMW(AtomicRMW* curr) {
 void WasmValidator::visitAtomicCmpxchg(AtomicCmpxchg* curr) {
   validateMemBytes(curr->bytes, curr->type, curr);
 }
-void WasmValidator::validateMemBytes(uint8_t bytes, WasmType ty, Expression* curr) {
+void WasmValidator::validateMemBytes(uint8_t bytes, WasmType type, Expression* curr) {
+  if (type == unreachable) {
```

Comment:
Stuff like `(i32.load (unreachable))` i.e., the pointer for the load is unreachable, so the load has type unreachable (like any node with an unreachable child).

(as before, this is normally removed by dce, but i was fuzzing with dce disabled on purpose)